code/modules/antagonists/changeling/powers/mutations.dm

```diff
@@ -244,30 +245,16 @@
 		chain = firer.Beam(src, icon_state = "tentacle", time = INFINITY, maxdistance = INFINITY, beam_sleep_time = 1)
 	..()
 
-/obj/item/projectile/tentacle/proc/reset_throw(mob/living/carbon/human/H)
-	if(H.in_throw_mode)
-		H.throw_mode_off() //Don't annoy the changeling if he doesn't catch the item
 
-/mob/proc/tentacle_grab(mob/living/carbon/C)
-	if(Adjacent(C))
-		var/obj/item/grab/G = C.grabbedby(src,1)
-		if(istype(G))
-			G.state = GRAB_AGGRESSIVE //Instant aggressive grab
 
 /mob/proc/tentacle_stab(mob/living/carbon/C)
 	if(Adjacent(C))
 		var/obj/item/I = r_hand
-		if(!is_sharp(I))
+		if(I == null || istype(/obj/item/gun/magic/tentacle, I))
 			I = l_hand
-		if(!is_sharp(I))
+		if(I == null || istype(/obj/item/gun/magic/tentacle, I))
 			return
-
-		C.visible_message("<span class='danger'>[src] impales [C] with [I]!</span>", "<span class='userdanger'>[src] impales you with [I]!</span>")
-		add_attack_logs(src, C, "[src] pulled [C] with a tentacle, attacking them with [I]") //Attack log is here so we can fetch the item they're stabbing with.
-		C.apply_damage(I.force, BRUTE, "chest")
-		do_item_attack_animation(C, used_item = I)
-		add_blood(C)
-		playsound(get_turf(src), I.hitsound, 75, 1)
+		I.melee_attack_chain(src, C)//Hits the victim with whatever they are holding that is no the zero force tentacle
```

Comment:
What in gods name is this? Just shove the return in the above block please.
Also instead of `I == null` please use `!I`

Reply:
Not sure what return you want shoved but !I done